tests/local_test_transformer.py

```diff
@@ -0,0 +1,550 @@
+        self.assertTrue(x.equals(x_from_fit))
+
+
+class TestVeryManyFeatures(unittest.TestCase):
+    def __init__(self, methodName: str = None) -> None:
+        super().__init__(methodName)
+        self.rng = np.random.default_rng(1023884)
+        self.X_np = None
+        self.y_np = None
+        self.X_pd = None
+        self.y_pd = None
+
+    def create_data_numpy(self) -> None:
+        """
+        Idempotent function that instantiates a class variable containing test numpy data
+        """
+
+        if self.X_np is None:
+            self.X_np = self.rng.uniform(-10, 10, size=(10000, 25000))
+
+        if self.y_np is None:
+            self.y_np = np.array(
+                [int(i) for i in (self.rng.uniform(0, 1, size=(10000, 1)) > 0.5)]
+            )
+
+        return None
+
+    def create_data_pd(self) -> None:
+        """
+        Idempotent function that instantiates a class variable containing test pandas data
+        derived from the numpy data. If `create_data_numpy` has not been called, this function
+        will call it.
+        """
+        self.create_data_numpy()
+
+        if self.X_pd is None:
+            self.X_pd = pd.DataFrame(self.X_np)
+
+        if self.y_pd is None:
+            self.y_pd = pd.DataFrame(self.y_np)
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.create_data_pd()
+
+    def test_fit_transform_no_y(self):
+        """
+        Test the transform method without the outcome variable specified
+        """
+        logging.getLogger().setLevel(logging.INFO)
+
+        selector = SelectFromQuadraticModel(time_limit=30)
+
+        # test numpy without fit
+        x = selector.fit_transform(self.X_np, number_of_features=20)
+
+        self.assertEqual(x.shape[1], 20)
+
+        x_from_fit = self.X_np[:, selector.selected_columns]
+
+        np.testing.assert_array_equal(x, x_from_fit)
+
+        # test pandas withoput fit
+        x = selector.fit_transform(self.X_pd, number_of_features=20)
+
+        self.assertEqual(x.shape[1], 20)
+
+        x_from_fit = self.X_pd.loc[:, selector.selected_columns]
+        self.assertTrue(x.equals(x_from_fit))
+
+    def test_fit_transform_y(self):
+        """
+        Test the transform method with the outcome variable specified
+        """
+        logging.getLogger().setLevel(logging.INFO)
+
+        selector = SelectFromQuadraticModel(time_limit=30)
+
+        # test numpy without fit
+        x = selector.fit_transform(self.X_np, self.y_np, number_of_features=20)
+
+        self.assertEqual(x.shape[1], 20)
+
+        x_from_fit = self.X_np[:, selector.selected_columns]
+        np.testing.assert_array_equal(x, x_from_fit)
+
+        # test pandas withoput fit
+        x = selector.fit_transform(self.X_pd, self.y_pd, number_of_features=20)
+
+        self.assertEqual(x.shape[1], 20)
+
+        x_from_fit = self.X_pd.loc[:, selector.selected_columns]
+        self.assertTrue(x.equals(x_from_fit))
+
+
+if __name__ == "__main__":
```

Comment:
Not generally a good idea to include these, as they make assumptions about import path.